### PR TITLE
Automate UI test that creates new Quarkus project and tests whether the runtime has been downloaded

### LINKED ISSUE #282: Automate UI test that creates new Quarkus project and tests whether the runtime has been downloaded
- the test creates new Quarkus project using the **New Project -> Java -> Quarkus (Additional library) -> download**
- the test checks whether the runtime has been downloaded

### PR DESCRIPTION
**Pull request by zcervink**
Fixes #282

Signed-off-by: Zbynek Cervinka <zcervink@redhat.com>

### CODE REVIEW

**Comment by zcervink** [comment]
@jrichter1 I have created new test that tests whether the Quarkus runtime with specific version has been successfully downloaded and included in the project hierarchy. The whole test runs as expected, but at the end of the test there is a **ConnectException**. This exceptions appear after the IDE has been closed by clicking on the cross button in the top window corner. I have tried other options to close the IDE (such as key shortcut etc.), nothing helped.


### Terminal output:

```
Zbynek-MacBook-Pro:intellij-quarkus zcervink$ ./gradlew clean runIdeForUiTests -PideaVersion=2020.1& ./gradlew integrationTest
[1] 44977
Starting a Gradle Daemon, 1 busy and 3 stopped Daemons could not be reused, use --status for details

> Task :compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :integrationTest

org.jboss.tools.intellij.quarkus.BasicTests

  Test testIfTheQuarkusRuntimeIsDownloaded() PASSED (1m 11s)
  Test executionError FAILED

  java.net.ConnectException: Failed to connect to /127.0.0.1:8082
      at org.jboss.tools.intellij.quarkus.BasicTests.quitTheIde(BasicTests.java:45)
  Caused by: java.net.ConnectException: Connection refused (Connection refused)
      at org.jboss.tools.intellij.quarkus.BasicTests.quitTheIde(BasicTests.java:45)


FAILURE: Executed 2 tests in 1m 19s (1 failed)


2 tests completed, 1 failed

> Task :integrationTest FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':integrationTest'.
> There were failing tests. See the report at: file:///Users/zcervink/git/intellij-quarkus/build/reports/tests/integrationTest/index.html

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.3/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 2m 9s
12 actionable tasks: 9 executed, 3 up-to-date

[1]+  Stopped                 ./gradlew clean runIdeForUiTests -PideaVersion=2020.1

```


### Exception stacktrace (report from the file:///Users/zcervink/git/intellij-quarkus/build/reports/tests/integrationTest/index.html mentioned in the terminal output above):

```
java.net.ConnectException: Failed to connect to /127.0.0.1:8082
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:265)
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:183)
	at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.java:224)
	at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.java:108)
	at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.java:88)
	at okhttp3.internal.connection.Transmitter.newExchange(Transmitter.java:169)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:41)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:88)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:229)
	at okhttp3.RealCall.execute(RealCall.java:81)
	at retrofit2.OkHttpCall.execute(OkHttpCall.java:188)
	at com.intellij.remoterobot.client.IdeRobotClient.execute(IdeRobotClient.kt:62)
	at com.intellij.remoterobot.JavaScriptApi$DefaultImpls.runJs(JavaScriptApi.kt:53)
	at com.intellij.remoterobot.RemoteRobot.runJs(RemoteRobot.kt:26)
	at com.intellij.remoterobot.fixtures.Fixture.runJs(Fixture.kt:73)
	at com.intellij.remoterobot.fixtures.Fixture.runJs$default(Fixture.kt:71)
	at com.intellij.remoterobot.fixtures.Fixture.runJs(Fixture.kt)
	at org.jboss.tools.intellij.quarkus.utils.GlobalUtils.lambda$quitIntelliJFromTheWelcomeDialog$9(GlobalUtils.java:151)
	at com.intellij.remoterobot.stepsProcessing.StepWorkerKt$step$1.invoke(StepWorker.kt:14)
	at com.intellij.remoterobot.stepsProcessing.StepWorkerKt$step$1.invoke(StepWorker.kt)
	at com.intellij.remoterobot.stepsProcessing.StepWorkerKt.step(StepWorker.kt:21)
	at com.intellij.remoterobot.stepsProcessing.StepWorkerKt.step(StepWorker.kt:13)
	at org.jboss.tools.intellij.quarkus.utils.GlobalUtils.quitIntelliJFromTheWelcomeDialog(GlobalUtils.java:148)
	at org.jboss.tools.intellij.quarkus.BasicTests.quitTheIde(BasicTests.java:45)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:515)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.lambda$invokeAfterAllMethods$9(ClassTestDescriptor.java:387)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:72)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.lambda$invokeAfterAllMethods$10(ClassTestDescriptor.java:386)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1080)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.invokeAfterAllMethods(ClassTestDescriptor.java:386)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.after(ClassTestDescriptor.java:217)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.after(ClassTestDescriptor.java:74)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:119)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:72)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:119)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:71)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:110)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:72)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:71)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:99)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:79)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:75)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:61)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.stop(TestWorker.java:132)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:412)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at okhttp3.internal.platform.Platform.connectSocket(Platform.java:130)
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:263)
	... 93 more
```

**Comment by zcervink** [comment]
@jrichter1 maybe I do not need your help after all - this issue has been resolved just by not quitting IntelliJ. Maybe we can further investigate this issue, but I think it is not necessary right now - the second process with tests just ends with tests results. Well but I see the build failed on Linux. I have discussed something similar with @jeffmaury some time ago - Jeff do you think I should comment the test as I have done before? Or just leave it as it is and fix it sometimes later?

**Details of the failed test:**
```
com.redhat.devtools.intellij.quarkus.completion.MavenApplicationPropertiesCompletionTest

  Test testBooleanCompletion FAILED (12.1s)

  junit.framework.AssertionFailedError: expected:<2> but was:<0>
      at com.redhat.devtools.intellij.quarkus.completion.MavenApplicationPropertiesCompletionTest.testBooleanCompletion(MavenApplicationPropertiesCompletionTest.java:41)
```

**Comment by jeffmaury** [comment]
I prefer to keep it, and I have an idea on how to fix it, will submit a separate PR

**Comment by zcervink** [comment]
Please review @jrichter1 @jeffmaury. To @jrichter1: it should be working on Linux now.

**Comment by jeffmaury** [comment]
@zcervink I introduced my changes from #258 but UI tests are still failing

**Comment by zcervink** [comment]
@jeffmaury I have fixed the issue with the empty List, now all checks are green. Please review.

**Comment by zcervink** [comment]
@jeffmaury @jrichter1 After many many trials I finally come to the solution. Github actions (linux) are now working as well as the UI tests. All checks are green I think, the PR is ready for review.

File: src/it/java/org/jboss/tools/intellij/quarkus/utils/QuarkusUtils.java [hunk 1]
```diff
@@ -76,4 +79,30 @@ public static void selectQuarkusExtensions(NewProjectDialogFixture newProjectDia
...
+    public static String createNewJavaProjectWithQuarkusFramework(RemoteRobot remoteRobot, String projectName) {
+        final WelcomeFrameDialogFixture welcomeFrameDialogFixture = remoteRobot.find(WelcomeFrameDialogFixture.class);
+        welcomeFrameDialogFixture.createNewProjectLink().click();
+        final NewProjectDialogFixture newProjectDialogFixture = welcomeFrameDialogFixture.find(NewProjectDialogFixture.class, Duration.ofSeconds(20));
+        NewProjectDialogUtils.selectNewProjectType(remoteRobot, "Java");
+
+        newProjectDialogFixture.theFrameworksTree().findText("Quarkus").click();
+        Point quarkusCheckboxLocation = newProjectDialogFixture.theFrameworksTree().findText("Quarkus").getPoint();
+        quarkusCheckboxLocation.x = 10;
+        newProjectDialogFixture.theFrameworksTree().click(quarkusCheckboxLocation);
```
Comment (jrichter1) [inline src/it/java/org/jboss/tools/intellij/quarkus/utils/QuarkusUtils.java:92]: Is there no way to actually find the checkbox ?

Comment (zcervink) [reply src/it/java/org/jboss/tools/intellij/quarkus/utils/QuarkusUtils.java:92]: There is no way to find the checkbox easily right now. The checkboxes are not accessible using the Remote-Robot, they are wrapped inside "ExtensionsTable" element. I suggest to leave it for now as it is. See:

Snímek obrazovky 2021-02-02 v 23 47 57

Snímek obrazovky 2021-02-02 v 23 55 32

File: src/it/java/org/jboss/tools/intellij/quarkus/utils/GlobalUtils.java [hunk 2]
```diff
@@ -263,7 +267,7 @@ private static int getNumberOfProjectLinks(RemoteRobot remoteRobot) {
     }
 
     public static void waitUntilIntelliJStarts(int port) {
-        waitFor(Duration.ofSeconds(300), Duration.ofSeconds(3), "The IntelliJ Idea did not start in 5 minutes.", () -> isIntelliJUIVisible(port));
+        waitFor(Duration.ofSeconds(1800), Duration.ofSeconds(3), "The IntelliJ Idea did not start in 30 minutes.", () -> isIntelliJUIVisible(port));
```
[the remark this answers is not in the document]
Comment (zcervink) [reply src/it/java/org/jboss/tools/intellij/quarkus/utils/GlobalUtils.java]: Set to 10 minutes, just for sure.

File: src/it/java/org/jboss/tools/intellij/quarkus/BasicTests.java [hunk 3]
```diff
@@ -63,4 +58,19 @@ public void createAQuarkusProjectAndBuildItUsingMaven() {
...
+        step("Test whether the Quarkus runtime can be downloaded", () -> {
+            String projectName = "java-project-with-quarkus-runtime";
+            String runtimeJarName = QuarkusUtils.createNewJavaProjectWithQuarkusFramework(robot, projectName);
+            GlobalUtils.waitUntilTheProjectImportIsComplete(robot);
+            GlobalUtils.closeTheTipOfTheDayDialogIfItAppears(robot);
+            GlobalUtils.maximizeTheIdeWindow(robot);
+            GlobalUtils.waitUntilAllTheBgTasksFinish(robot);
+            assertTrue(ProjectToolWindowUtils.isAProjectFilePresent(robot, projectName, "lib", runtimeJarName), "The runtime has not been downloaded.");
+            GlobalUtils.checkForExceptions(robot);
+            GlobalUtils.closeTheProject(robot);
```
[the remark this answers is not in the document]
Comment (zcervink) [reply src/it/java/org/jboss/tools/intellij/quarkus/BasicTests.java]: Well some future tests maybe do not need this actions, but for now I have moved it to AfterEach.

File: src/it/java/org/jboss/tools/intellij/quarkus/utils/ProjectToolWindowUtils.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        }
+        return true;
+    }
+
+    private static void navigateThroughTheProjectTree(RemoteRobot remoteRobot, ActionToPerform action, String... pathArray) {
+        step("Navigate through the project tree", () -> {
+            List<String> path = Arrays.asList(pathArray);
+            Iterator<String> pathIterator = path.iterator();
+            while (pathIterator.hasNext()) {
+                final ProjectToolWindowFixture projectToolWindowFixture = remoteRobot.find(ProjectToolWindowFixture.class);
```
[the remark this answers is not in the document]
Comment (zcervink) [reply src/it/java/org/jboss/tools/intellij/quarkus/utils/ProjectToolWindowUtils.java:49]: No, it could not. It needs to be performed in every iteration, because the visible items changes in every iteration. The ProjectViewTree containing the file and folder/package names extends when you expand a folder/package. If it is performed outside the loop, the firstly hidden file/folder items are not accessible.

### See the "src" folder - it contains some other items. The items are hidden before expanding the "src" folder therefore they are not accessible using the RR framework right now:
Snímek obrazovky 2021-02-03 v 0 27 42

[the remark this answers is not in the document]
Comment (zcervink) [reply src/it/java/org/jboss/tools/intellij/quarkus/utils/ProjectToolWindowUtils.java:49]: No, it could not. It needs to be performed in every iteration, because the visible items changes in every iteration. The ProjectViewTree containing the file and folder/package names extends when you expand a folder/package. If it is performed outside the loop, the firstly hidden file/folder items are not accessible.

### See the "src" folder - it contains some other items. The items are hidden before expanding the "src" folder therefore they are not accessible using the RR framework right now:

Snímek obrazovky 2021-02-03 v 0 27 42

File: src/it/java/org/jboss/tools/intellij/quarkus/utils/ProjectToolWindowUtils.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            return false;
+        }
+        return true;
+    }
+
+    private static void navigateThroughTheProjectTree(RemoteRobot remoteRobot, ActionToPerform action, String... pathArray) {
+        step("Navigate through the project tree", () -> {
+            List<String> path = Arrays.asList(pathArray);
+            Iterator<String> pathIterator = path.iterator();
+            while (pathIterator.hasNext()) {
```
[the remark this answers is not in the document]
Comment (zcervink) [reply src/it/java/org/jboss/tools/intellij/quarkus/utils/ProjectToolWindowUtils.java]: Fixed -> rewritten to iterate through the array.

File: src/it/java/org/jboss/tools/intellij/quarkus/utils/ProjectToolWindowUtils.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (zcervink) [reply src/it/java/org/jboss/tools/intellij/quarkus/utils/ProjectToolWindowUtils.java]: UPDATE: It could be moved outside the loop. Fixed.

**Review by jrichter1 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -27,7 +27,30 @@ jobs:
...
+        name: ${{ runner.os }}-integration-test-reports
+        path: build/reports/tests/integrationTest
+      if: always()
+    - name: Archiving screenshots
+      uses: actions/upload-artifact@v2
+      with: 
+        name: screenshots
+        path: |
+          build/screenshots/*
+        retention-days: 5
```
Comment (jeffmaury) [inline .github/workflows/ci.yml:53]: Workflow runs have their own retention time why is this needed ?

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -27,7 +27,30 @@ jobs:
...
+    - name: Run integration tests
+      run: |
+        dir=~/.java/.userPrefs/jetbrains/_\!\(\!\!cg\"p\!\(\}\!\}@\"j\!\(k\!\|w\"w\!\'8\!b\!\"p\!\':\!e@=\=
+        mkdir -p "$dir"
+        cp .github/workflows/prefs.xml $dir
+        mkdir -p ~/.local/share/JetBrains/consentOptions
+        cp .github/workflows/accepted ~/.local/share/JetBrains/consentOptions
+        export DISPLAY=:99.0
+        Xvfb -ac :99 -screen 0 1920x1080x16 &
+        ./gradlew runIdeForUiTests -PideaVersion=2020.1 &
```
Comment (jeffmaury) [inline .github/workflows/ci.yml:40]: Shouldn't we target latest IJ version ie 2020.3 ?

File: src/it/java/org/jboss/tools/intellij/quarkus/utils/HelperUtils.java [hunk 8]
```diff
@@ -23,6 +23,10 @@
 public class HelperUtils {
 
     public static String listOfRemoteTextToString(List<RemoteText> data) {
+        if (data.isEmpty()) {
```
Comment (jeffmaury) [inline src/it/java/org/jboss/tools/intellij/quarkus/utils/HelperUtils.java:26]: Why is this required ?

File: .github/workflows/ci.yml (hunk 7 above)
Comment (zcervink) [reply .github/workflows/ci.yml:40]: Yes but the IDE has very different UI from 2020.1 (which is currently supported by the UI tests). I will perform updates of UI tests very soon in separated PR, I have jira for that already created - https://github.com/redhat-developer/intellij-quarkus/issues/317 (I do not want to make this PR too big and keep it stucked too long before merging).

File: .github/workflows/ci.yml (hunk 6 above)
Comment (zcervink) [reply .github/workflows/ci.yml:53]: I have copied this lines of code from @odockal. We came to the conclusion that 5 days is accurate period of time to keep the files. If you think I should delete it, it is ok for me.

**Copied from:**
(https://github.com/redhat-developer/vscode-server-connector/blob/master/.github/workflows/gh-actions.yml#L36)

File: src/it/java/org/jboss/tools/intellij/quarkus/utils/HelperUtils.java (hunk 8 above)
Comment (zcervink) [reply src/it/java/org/jboss/tools/intellij/quarkus/utils/HelperUtils.java:26]: Because I need to return a not null String from that method. I usually call some method to search inside the String etc. I see it more simple to check it inside the method than the every time outside the method. But if you see it more clearly implemented not to check it here, I will rewrite the tests.

Comment (jeffmaury) [reply src/it/java/org/jboss/tools/intellij/quarkus/utils/HelperUtils.java:26]: If you remove that piece of code but execute the result you will get an empty string as result,no ?

Comment (zcervink) [reply src/it/java/org/jboss/tools/intellij/quarkus/utils/HelperUtils.java]: Oh yes, you are right. I did understand the code wrong, after I have executed it now I finally see that it is not required -> removed.

File: .github/workflows/ci.yml (hunk 6 above)
Comment (zcervink) [reply .github/workflows/ci.yml:53]: @jeffmaury looks like this is the last issue - do you want me to delete the 'retention-days: 5 ' line or we just leave it as it is (and maybe delete is later)?

Comment (jeffmaury) [reply .github/workflows/ci.yml:53]: Remove it

Comment (zcervink) [reply .github/workflows/ci.yml]: Removed.

**Review by jeffmaury — APPROVED** [review]
LGTM
